### ISSUE TITLE
Change cleanup order in ScopedJob

### ISSUE THREAD

**Issue by otto-gebb** [AndyPook/QuartzHostedService#1]
Currently it's

```cs
public void Dispose()
{
    _scope.Dispose();
    (_innerJob as IDisposable)?.Dispose();
}
```

But maybe it should be

```cs
public void Dispose()
{
    (_innerJob as IDisposable)?.Dispose();
    _scope.Dispose();
}
```

to allow `_innerJob` to use its dependencies in its `Dispose` method.

**Comment by AndyPook** [comment]
indeed, thank you for taking the time to have a look

**Comment by AndyPook** [comment]
fixed by 2d8bd4645899b9fc7d4639492e9c2bfcebc7a2a5
